template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/spark/SparkOnYarnRoleConfigProvider.java

```diff
@@ -20,15 +24,23 @@ public class SparkOnYarnRoleConfigProvider extends AbstractRoleConfigProvider {
 
     private static final String SPARK_YARN_ACCESS_DIR_PARAM = "spark.yarn.access.hadoopFileSystems=";
 
+    private static final String SPARK_HADOOP_S3_SSL_CHANNEL_MODE = "spark.hadoop.fs.s3a.ssl.channel.mode=openssl";
+
     @Override
     protected List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, TemplatePreparationObject source) {
         switch (roleType) {
             case SparkRoles.GATEWAY:
-                return ConfigUtils.getStorageLocationForServiceProperty(source, HMS_METASTORE_EXTERNAL_DIR)
-                        .map(location -> ConfigUtils.getBasePathFromStorageLocation(location.getValue()))
-                        .map(basePath -> List.of(config(SPARK_CONF_CLIENT_SAFETY_VALVE,
-                                SPARK_YARN_ACCESS_DIR_PARAM + basePath)))
-                        .orElseGet(List::of);
+                List<ApiClusterTemplateConfig> roleConfigs = new ArrayList<>();
```

Comment:
Spark3OnYarn* also needs to change?

Reply:
openssl related configs are not present for SPARK3_ON_YARN

Reply:
Don't understand this ... openssl configs have been removed from the spark3 template? Would this set of changes take care of making them available for these templates as well? There's a separate Spark3OnYarn..Provider otherwise.

Reply:
Spark3 template was not updated to remove the config. I have updated templates cdp-rt-data-mart-spark3.bp in 7.2.16 and 7.2.17. Also, updated Spark3OnYarnRoleConfigProvider to populated them for AWS only similar to SparkOnYarnRoleConfigProvider